Polyhedron/demo/Polyhedron/Plugins/Operations_on_polyhedra/Diff_between_meshes_plugin.cpp

```diff
@@ -67,95 +68,12 @@ QList<QAction*> Polyhedron_demo_diff_between_meshes_plugin::actions() const {
   return QList<QAction*>() << actionDiff;
 }
 
-SMesh* Polyhedron_demo_diff_between_meshes_plugin::diff(SMesh* m1, SMesh* m2, bool compute_common = false)
-{
-  std::map<Point_3, std::size_t> point_id_map;
-  std::vector<std::size_t> m1_vertex_id(num_vertices(*m1), -1);
-  std::vector<std::size_t> m2_vertex_id(num_vertices(*m2), -1);
-
-  //iterate both meshes to set ids to all points, and set vertex/point_id maps.
-  std::size_t id =0;
-  for(auto v : m1->vertices())
-  {
-    Point_3 p = m1->point(v);
-    auto res = point_id_map.insert(std::make_pair(p, id));
-    if(res.second)
-      id++;
-    m1_vertex_id[(std::size_t)v]=res.first->second;
-  }
-  for(auto v : m2->vertices())
-  {
-    Point_3 p = m2->point(v);
-    auto res = point_id_map.insert(std::make_pair(p, id));
-    if(res.second)
-      id++;
-    m2_vertex_id[(std::size_t)v]=res.first->second;
-  }
-
-  //fill a set with the "faces point-ids" of m1 and then iterate faces of m2 to compare.
-  std::set<std::vector<std::size_t> > m1_faces;
-  for(auto f : m1->faces())
-  {
-    std::vector<std::size_t> ids;
-    for(auto v : CGAL::vertices_around_face(halfedge(f, *m1), *m1))
-    {
-      ids.push_back(m1_vertex_id[(std::size_t)v]);
-    }
-    std::sort(ids.begin(), ids.end());
-    m1_faces.insert(ids);
-  }
-
-  std::vector<SMesh::Face_index> common_faces;
-  std::vector<Point_3> common_points;
-  std::map<SMesh::Vertex_index, std::size_t> id_map;
-  id = 0;
-
-  for(auto f : m2->faces())
-  {
-    std::vector<std::size_t> ids;
-    for(auto v : CGAL::vertices_around_face(halfedge(f, *m2), *m2))
-    {
-      ids.push_back(m2_vertex_id[(std::size_t)v]);
-    }
-    std::sort(ids.begin(), ids.end());
-    if(!((m1_faces.find(ids) != m1_faces.end()) ^ compute_common))
-    {
-      common_faces.push_back(f);
-      for(auto v : CGAL::vertices_around_face(halfedge(f, *m2), *m2))
-      {
-        auto res = id_map.insert(std::make_pair(v,id));
-        if(res.second)
-        {
-          common_points.push_back(m2->point(v));
-          id++;
-        }
-      }
-    }
-  }
-
-  //iterate m1_faces and fill a polygon vector using the id_map previously filled.
-  std::vector<std::vector<std::size_t> > polygons(common_faces.size());
-  id = 0;
-  for(auto f : common_faces)
-  {
-    for(auto v : vertices_around_face(halfedge(f, *m2),*m2))
-    {
-      polygons[id].push_back(id_map[v]);
-    }
-    ++id;
-  }
-
-  SMesh* common = new SMesh();
-  CGAL::Polygon_mesh_processing::polygon_soup_to_polygon_mesh<SMesh>(
-    common_points, polygons, *common);
-  return common;
-
-}
-
 
 void Polyhedron_demo_diff_between_meshes_plugin::diff()
 {
 
+  typedef CGAL::Face_filtered_graph<SMesh> Filtered_graph;
```

Comment:
Note that Face filtered graph can be dangerous if your selection yelds non-manifold vertices. Maybe for copy_face_graph it is not an issue but it should be checked.